Auxiliary/DirectXTexXboxDetile.cpp

```diff
@@ -32,7 +32,9 @@ namespace
         const uint8_t* sptr = xbox.GetPointer();
         const uint8_t* endPtr = sptr + layout.SizeBytes;
 
-        for (uint32_t item = 0; item < nimages; ++item)
+        assert((nimages > 0) && (nimages <= UINT32_MAX));
```

Comment:
I've already checked this condition at the entry-point, but to make sure the assertion holds over time, I've added the assert.